<launch>
    <node name="pepperLocalization" pkg="cssr_system" type="pepperLocalization" output="screen">
        <!-- Marker and camera parameters -->
        <param name="marker_size" value="0.2" />
        <param name="camera_frame" value="camera_color_optical_frame" />
        <param name="base_frame" value="base_link" />
        <param name="world_frame" value="map" />
        
        <!-- Camera offset from robot base -->
        <param name="camera_offset_x" value="0.0" />
        <param name="camera_offset_y" value="0.0" />
        <param name="camera_offset_z" value="1.2" />
        
        <!-- Filtering parameters -->
        <param name="filter_alpha" value="0.3" />  <!-- 0.0 = heavy filtering, 1.0 = no filtering -->
        
        <!-- Debug parameters -->
        <param name="min_markers_for_pose" value="2" />
        <param name="max_detection_distance" value="8.0" />
    </node>
    
    <!-- RealSense camera -->
    <include file="$(find realsense2_camera)/launch/rs_camera.launch">
        <arg name="align_depth" value="true"/>
        <arg name="enable_color" value="true"/>
        <arg name="enable_depth" value="true"/>
        <arg name="color_width" value="1280"/>
        <arg name="color_height" value="720"/>
        <arg name="color_fps" value="30"/>
    </include>
</launch>